File: web/src/layout/ComponentTestXml.ts
//
// ComponentTestXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './component_test.html'
//! Declares com.lightningkite.butterflytemplate.layouts.ComponentTestXml
export class ComponentTestXml {
    xmlRoot!: HTMLElement
    label!: HTMLElement
    button!: HTMLButtonElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.label = getViewById<HTMLElement>(view, "label")
        this.button = getViewById<HTMLButtonElement>(view, "button")
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/ComponentTestXml.kt
//
// ComponentTestXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class ComponentTestXml {

    lateinit var label: TextView
    lateinit var button: Button
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.component_test, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        label = view.findViewById<TextView>(R.id.label)
        button = view.findViewById<Button>(R.id.button)
        
        
        
        return view
    }
}